### DISCUSSION TITLE
Solicitudes con estatus pending

### DISCUSSION THREAD

**Discussion by jparadaa** [AndresRamos/ARSoftware.Contpaqi.Comercial.Api#8] [Q&A] [answered]
Cómo se resuelve cuando un request devuelve un response con estatus "pending"

Saludos,
Javier

**Comment by AndresRamos** [comment, marked as answer]
Cuando envías una solicitud a la API, está la crea en la base de datos y espera 10 segundos para que el sincronizador la procese en Comercial, y te regresa el `Request` con el `Response`.

Si durante el tiempo de espera la solicitud no se procesó completamente, ya sea porque el sincronizador no está activo o porque tardo más del tiempo de espera en procesar la solicitud, la API te regresa el `Request` con los valores
* `"id": "421bc492-b1df-472d-320c-08db200c2937"` - Se le asigna un Id a la solicitud
* `"status": "Pending"` - Se queda con el estatus pendiente
* `"response": null` - El response se queda nulo

```json
{
    "id": "421bc492-b1df-472d-320c-08db200c2937",
    "subscriptionKey": "00000000000000000000000000000000",
    "empresaRfc": "URE180429TM6",
    "dateCreated": "2023-03-08T13:34:39.4362833",
    "status": "Pending",
    "response": null
}
```

Debera tomar el `id` y consultar su estatus en el endpoint GET `/api/Requests/{id}`

**Comment by jparadaa** [comment]
Qué estatus devuelve la llamada al API en caso de que quede en pending?, es decir, status 201 - documento creado, 400 - solicitud inválida, pero qué status devuelve cuando se queda en pending?, y otra pregunta, sólo y únicamente en caso que quede en pending el response se genera únicamente con 

id:
status:
response:

**Reply by AndresRamos** [reply]
Si la solicitud se creó exitosamente **(en la base de datos)** la API te devuelve el `Status: 201` y te retorna el `Request` con un `Id` nuevo y el `status` = `Pending` o `Processed`.

`Pending` quiere decir que la solicitud se creó en la base de datos, pero está pendiente por ser procesada en Comercial por el sincronizador.

`Processed` quiere decir que la solicitud ya fue procesada en Comercial.

Cuando tu envías una solicitud a la API sucede lo siguiente:
1. Se valida la solicitud
2. Se crea la solicitud en la base de datos y se asigna su status a `pending`
3. Se espera un máximo de 10 segundos para ser procesada por el sincronizador
4. Te regresa la solicitud creada con su `Id` asignado
    1. Si la solicitud se pudo procesar durante el periodo de espera (los 10s), la solicitud tendrá un status `Processed` y el objecto `Response` estará instanciado con los valores de la respuesta correspondiente.
    2. Si la solicitud no se pudo procesar durante el periodo de espera (los 10s), la solicitud tendrá un status `Pending` y el objecto `Response` sera `null`

**Comment by jparadaa** [comment]
Con la actualización queda más entendible al menos para mi.
